python/simpler/mpi_direct_supervisor.py

```
@@ -0,0 +1,375 @@
+            f"{mpi4py_vendor} ({mpi4py_family}); use a matching launcher or rebuild "
+            "mpi4py with the launcher's MPI compiler"
+        )
+    session_id = secrets.randbits(64) or 1
+    manifest = topology.runtime_manifest(session_id)
+    manifest_json = base64.urlsafe_b64encode(
+        json.dumps(manifest, separators=(",", ":")).encode("utf-8")
+    ).decode("ascii")
+    gate_token = secrets.token_urlsafe(32)
+    listener = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    listener.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+    listener.bind(("0.0.0.0", 0))
```

Ruff (0.16.1)

[error] Possible binding to all interfaces

(S104)